apps/web/src/components/Notification/Type/MentionNotification.tsx

```diff
@@ -16,9 +19,25 @@ interface Props {
   notification: NewMentionNotification;
 }
 
+const messages: Record<string, MessageDescriptor> = {
+  comment: defineMessage({
+    id: '<0><1/> mentioned you in a <2>comment</2></0>'
```

Comment:
I just want clarification on how this works 🙇🏼

Reply:
Sure :) First, the reason why there are three explicit sentences, and not just one (something like
`username +  "mentioned you in a"  + publication.typename`)
is that many languages will have different sentence structure depending on parameters. For example in Swedish (and many other languages), the translation of "your" is dependent on whether it's "your comment", "your post", etc.
So the developers should always try to avoid building sentences out of smaller parts in code.
If needed, we can add some guidelines for developers, how to write translatable code.

As for the technical explanation, the `defineMessage` is a macro that tells LinguiJS that this string exists, and to add it to .po source file. LinguiJS is not intelligent enough to extract source strings for concatenated strings (such as `"Upload your" + t(obj.typename)`) so any time there are multiple possible sentences like this, we need to define those strings like this.

Lastly, the `defaultMessage` is for future compatibility, if another publication type is added, we will at least get a default message in English.

Not sure if this answers your question but I hope so :)

Reply:
makes sense, thanks 🙏🏼